UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/FileShareService.cs

```diff
@@ -247,6 +247,75 @@ private static async Task CopyFileToFolder(HttpResponseMessage httpResponse, str
+            httpReadMeFileResponse = await fileShareServiceClient.CallFileShareServiceApi(HttpMethod.Get, payloadJson, accessToken, readMeFilePath);
+            if (httpReadMeFileResponse.IsSuccessStatusCode)
+            {
+                using (Stream stream = await httpReadMeFileResponse.Content.ReadAsStreamAsync())
+                {
+                    using (var outputFileStream = new FileStream(file, FileMode.Create, FileAccess.ReadWrite))
+                    {
+                        stream.CopyTo(outputFileStream);
+                    }
+                    using StreamReader reader = new StreamReader(stream);
+                    secondLineText = GetLine(file, line);
+                }
+                string text = File.ReadAllText(file);
+                text = secondLineText.Length == 0 ? lineToWrite: text.Replace(secondLineText, lineToWrite);
+                File.WriteAllText(file, text);               
+            }
+            else
+                logger.LogInformation(EventIds.ReadMeTextFileIsNotDownloaded.ToEventId(), "Error in downloading readme.txt file for {BatchId}", batchId);
+
+            return true;
+        }
+
+        public async Task<string> SearchReadMeFilePath(string batchId)
+        {
+            string payloadJson = string.Empty;
+            var accessToken = await authTokenProvider.GetManagedIdentityAuthAsync(fileShareServiceConfig.Value.ResourceId);
+            string filePath = string.Empty;           
+            var uri = $"{fileShareServiceConfig.Value.BaseUrl}/batch?$filter={fileShareServiceConfig.Value.ProductType} fileName eq '{fileShareServiceConfig.Value.ReadMeFileName}' and BusinessUnit eq '{fileShareServiceConfig.Value.BusinessUnit}'";
+            HttpResponseMessage httpResponse;
+            httpResponse = await fileShareServiceClient.CallFileShareServiceApi(HttpMethod.Get, payloadJson, accessToken, uri);
+            if (httpResponse.IsSuccessStatusCode)
+            {
+                SearchBatchResponse searchBatchResponse = await SearchBatchResponse(httpResponse);
+                if (searchBatchResponse.Entries.Count > 0)
+                {
+                    var batchResult = searchBatchResponse.Entries.FirstOrDefault();
+                    filePath =  batchResult.Files.FirstOrDefault().Links.Get.Href;                  
+                }
+                else
+                    logger.LogInformation(EventIds.ReadMeTextFileNotFound.ToEventId(), "Readme.txt file not found for {BatchId}", batchId);
+            }
+            else
+                logger.LogInformation(EventIds.QueryFileShareServiceNonOkResponse.ToEventId(), "Query File share service for readme file with uri {RequestUri} responded with {StatusCode} for {BatchId}", httpResponse.RequestMessage.RequestUri, httpResponse.StatusCode, batchId);
+
+            return filePath;
+        }
+
+        private static string GetLine(string filePath, int line)
+        {
+            string secondLine = string.Empty;
```

Comment:
Would this variable be better names as lineToFind (or something similar)?

I realise in this case we are looking for the second line but in reality this is a generic method to find the line of the number passed in.

Reply:
or perhaps lineFound?

Reply:
Done. Modified method.